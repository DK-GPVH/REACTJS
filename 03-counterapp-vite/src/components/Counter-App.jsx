import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value})=>{
    const [ Counter,setCounter] = useState(value);

    const add =()=>{
        // console.log("hola");
        //setCounter(Counter + 1);
        setCounter(Counter+1);
    };
    const sub = ()=>{
        setCounter(Counter-1);
    };
    const reset = ()=>{
        setCounter(value);
    }
    return <>
        <h1>CounterAPP</h1>
        <h2>{ Counter}</h2>

        <button onClick={add}> ADD +</button>
        <button onClick={sub}> SUBTRACT -</button>
        <button aria-label='btn-reset' onClick={reset}> RESET Â°</button>

    </>
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}

CounterApp.defaultProps = {
    value : "No enviaste ningun valor"
}
