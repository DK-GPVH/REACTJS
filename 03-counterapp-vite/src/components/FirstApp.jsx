import PropTypes from "prop-types";

const objeto = {
    nombre : "Jorge",
    email : "@gmail.com",
    edad : 45
}

function funcion () {
    return console.log("hola");
}
export const FirstApp=(props)=>{
    //<Fragment> => <>
    return (
        //no se puede mandar objetos {}
        //para enviar objetos mediante stringify
        <>
        {/* <h1>{"Hola"}FirstApp</h1> */}
        {/* <p>{props.title}</p>
        <h1>{props.title}</h1>
        <h2>{JSON.stringify(objeto)}</h2>
        <h3>{funcion()}</h3>         */}
        {/*crando un data-testid*/}
        <h1 data-testid="test-title">{props.title}</h1>
        <p>{props.subtitle}</p>
        <p>{props.nombre}</p>
        </> 
        

    )
};

//definir los props por defecto
FirstApp.defaultProps = {
    subtitle:'No hay subtitulo',
    nombre:'no hay nombre',
    // title : 'No hay titulo',
}

//Atributos de los props
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string
}

