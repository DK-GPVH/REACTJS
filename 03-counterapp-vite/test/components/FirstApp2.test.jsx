import {render,screen} from '@testing-library/react';
import {FirstApp} from '../../src/components/FirstApp';

describe('Pruebas en firstApp mas optimizada',()=>{
    const title="Titulo de prueba";
    const subtitle = "Subtitulo de prueba";
    test('DEbe hacer match con el snapshot',()=>{
        const {container} =render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot(); 
    });

    test('Debe mostrarr el mensaje',()=>{
        render(<FirstApp title={title}/>);
        // screen.debug()
        //uso del screen
        expect(screen.getByText(title)).toBeTruthy();
    });

    test('Debe mostrar el titulo en un h1',()=>{
        render(<FirstApp title={title}/>);
        expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('Debe mostrar el subtitulo enviado por props',()=>{
        render(<FirstApp title={title} subtitle={subtitle}/>);
        expect(screen.getAllByText(subtitle).length).toBe(1);
    });
});