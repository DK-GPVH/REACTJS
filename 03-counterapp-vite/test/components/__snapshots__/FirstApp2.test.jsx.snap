// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en firstApp mas optimizada DEbe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Titulo de prueba
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    no hay nombre
  </p>
</div>
`;
